homeassistant/components/sonos/speaker.py

```diff
@@ -456,6 +456,34 @@ def async_dispatch_favorites(self, event: SonosEvent) -> None:
     @callback
     def async_dispatch_media_update(self, event: SonosEvent) -> None:
         """Update information about currently playing media from an event."""
+        # The new coordinator can be provided in a media update event but
+        # before the ZoneGroupState updates. If this happens the playback
+        # state will be incorrect and should be ignored. Switching to the
+        # new coordinator will use its media. The regrouping process will
+        # be completed during the next ZoneGroupState update.
+        av_transport_uri = event.variables.get("av_transport_uri", "")
+        current_track_uri = event.variables.get("current_track_uri", "")
+        if av_transport_uri == current_track_uri and av_transport_uri.startswith(
+            "x-rincon:"
+        ):
+            new_coordinator_uid = av_transport_uri.split(":")[-1]
```

Comment:
Is there ever a chance we will get one of these and the coordinator will be the same?

Reply:
I guess it really doesn't matter either way since it will still be the right one even if it's the same